### PR TITLE
Minimize the build graph by deleting unnecessary references

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/runtime/issues/65552

Minimizing projects' dependency graph. There are tons of unnecessary
Reference and ProjectReference items which aren't required anymore as
the referenced projects became full façade assemblies. Removing those
from a leaf's graph makes the graph smaller and therefore the project's
evaluation and build faster.

Examples of full facade projects are:
- System.Runtime.CompilerServices.Unsafe.csproj
- System.Runtime.Extensions.csproj

The changes were performed via:
* Find & Replace of:
  * `<Reference Include="full-facade-assembly" />`
  * `<ProjectReference Include="full-facade-project" />`
* Manually checking if the compiled assembly actually requires the
defined ProjectReference/Reference items and removing them if not.

Unrelated changes that I applied in the projects that I already touched:
* In a handful of projects, adding missing primary references that were
previously brought in transitively.
* Adding empty lines between property and/or item groups to make
project files more readable. Most projects in src/libraries already
follow that style.
* Moving a few Compile items into their own ItemGroup to improve
readability of project files.

### CODE REVIEW

File: src/libraries/System.Diagnostics.TraceSource/src/System.Diagnostics.TraceSource.csproj [hunk 1]
```diff
@@ -34,8 +34,9 @@
     <ProjectReference Include="$(LibrariesProjectRoot)System.Collections.Specialized\src\System.Collections.Specialized.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.ComponentModel.Primitives\src\System.ComponentModel.Primitives.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime\src\System.Runtime.csproj" />
-    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.Extensions\src\System.Runtime.Extensions.csproj" />
     <ProjectReference Include="$(LibrariesProjectRoot)System.Threading\src\System.Threading.csproj" />
+    <!-- error : CannotChangeAttribute : Attribute 'System.Diagnostics.SwitchLevelAttribute' on 'System.Diagnostics.BooleanSwitch' changed
+         from '[SwitchLevelAttribute(typeof(bool))]' in the implementation to '[SwitchLevelAttribute(typeof(Boolean))]' in the reference. -->
     <Reference Include="System.IO.FileSystem" />
```
Comment (ViktorHofer) [inline src/libraries/System.Diagnostics.TraceSource/src/System.Diagnostics.TraceSource.csproj:40]: @tannergooding do you think the above mentioned ApiCompat issue will be fixed with your recent change to use primitive types always?

File: src/libraries/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj [hunk 2]
```diff
@@ -260,7 +260,7 @@ System.Reflection.PortableExecutable.ManagedPEBuilder</PackageDescription>
     <ProjectReference Include="$(LibrariesProjectRoot)System.Collections.Immutable\src\System.Collections.Immutable.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFrameworkVersion)' == 'v$(NetCoreAppCurrentVersion)'">
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
```
Comment (ViktorHofer) [inline src/libraries/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj:263]: @eerhardt I'm applying [your feedback](https://github.com/dotnet/runtime/pull/68543#discussion_r859940692) here.

**Review by bartonjs — APPROVED** [review]
Assuming it builds, LGTM.

**Comment by ViktorHofer** [comment]
Requesting feedback from a few folks as this is general clean-up / build optimization and not specific to infrastructure.

File: src/libraries/System.Security.SecureString/ref/System.Security.SecureString.cs [hunk 3]
```diff
@@ -1,5 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (eerhardt) [inline src/libraries/System.Security.SecureString/ref/System.Security.SecureString.cs:1]: @GrabYourPitchforks sees "delete SecureString.cs" and starts to celebrate 🥳

File: src/libraries/System.Diagnostics.TraceSource/src/System.Diagnostics.TraceSource.csproj (hunk 1 above)
Comment (tannergooding) [reply src/libraries/System.Diagnostics.TraceSource/src/System.Diagnostics.TraceSource.csproj:40]: This one I'm not sure is impacted. The logic I changed was only impacting `Boolean` itself using `bool` vs `Boolean` (and same for other primitives).

I wouldn't have expected it to also impact attribute decisions like this one. So there may still be another GenApi fix needed to ensure `typeof` here is allowing the keyword where possible.

File: src/libraries/System.Net.Quic/ref/System.Net.Quic.csproj [hunk 4]
```diff
@@ -5,14 +5,16 @@
...
-    <ProjectReference Include="..\..\System.Runtime\ref\System.Runtime.csproj" />
-    <ProjectReference Include="..\..\System.Net.Primitives\ref\System.Net.Primitives.csproj" />
-    <ProjectReference Include="..\..\System.Net.Sockets\ref\System.Net.Sockets.csproj" />
-    <ProjectReference Include="..\..\System.Net.Security\ref\System.Net.Security.csproj" />
-    <ProjectReference Include="..\..\System.Security.Cryptography.X509Certificates\ref\System.Security.Cryptography.X509Certificates.csproj" />
+    <ProjectReference Include="$(LIbrariesProjectRoot)System.Memory\ref\System.Memory.csproj" />
+    <ProjectReference Include="$(LIbrariesProjectRoot)System.Net.Primitives\ref\System.Net.Primitives.csproj" />
+    <ProjectReference Include="$(LIbrariesProjectRoot)System.Net.Security\ref\System.Net.Security.csproj" />
+    <ProjectReference Include="$(LIbrariesProjectRoot)System.Runtime\ref\System.Runtime.csproj" />
+    <ProjectReference Include="$(LIbrariesProjectRoot)System.Security.Cryptography\ref\System.Security.Cryptography.csproj" />
```
Comment (bartonjs) [inline src/libraries/System.Net.Quic/ref/System.Net.Quic.csproj:18]: Nit: the changes from ..\.. to a variable in this file have an inappropriate capital-I.

```suggestion
    <ProjectReference Include="$(LibrariesProjectRoot)System.Security.Cryptography\ref\System.Security.Cryptography.csproj" />
```

**Pull request merged**